hyperspy/io.py

```diff
@@ -771,6 +772,14 @@ def save(filename, signal, overwrite=None, **kwds):
             f"Please try one of {strlist2enumeration(yes_we_can)}"
         )
 
+    if writer.non_uniform_axis is False and nua is False:
```

Comment:
```suggestion
    if not writer.non_uniform_axis and not signal.axes_manager.all_uniform:
```
and remove `nua = signal.axes_manager.all_uniform` above to improve readability?